### DISCUSSION TITLE
Have another npm package depend on jbrowse2?

### DISCUSSION THREAD

**Discussion by bbimber** [GMOD/jbrowse-components#1474] [General]
Hello - are there any examples of another projects that depend on jbrowse2, and are their package.json files visible? Thanks in advance.

**Comment by rbuels** [comment]
Sounds like you are wanting to embed JB2 in another product, is that the
case?

**Comment by bbimber** [comment]
@rbuels Roughly. My group has a LabKey Server module that is sort of a wrapper around JBrowse 1. LabKey provides sort of a LIMS to track genomes, sequences, tracks, etc. The user can pick resources and create a jbrowse session on-the-fly. There is code in this module that either creates the trackList.json files, and/or calls the old perl scripts to process resources.

The module is here: https://github.com/BimberLab/DiscvrLabKeyModules/tree/discvr-20.7/jbrowse

I'm very interested in moving to JB2, since I'm hopeful we can do ever more to integrate with this code, such as using the config JSON of the JB2 session to drive other UI. This could include rendering features as a table, or rendering the search box as a standalone component.

**Comment by garrettjstevens** [comment]
JBrowse 2 is a standalone application, it's not really something that another package can "depend" on like a library. Therefore, JBrowse 2 is not published on NPM. Many of the pieces used to build JBrowse 2 are published on NPM under the `@jbrowse` namespace (https://www.npmjs.com/org/jbrowse), but it doesn't sound like that would work for your use case.

Also, we are working on a React component that will be published on NPM that contains a single linear genome view from JBrowse 2, but that wouldn't have the full app experience.

I'm not familiar with LabKey Server, so I'm not totally sure how your current extension works, but we do have the JBrowse CLI ([`@jbrowse/cli`](https://www.npmjs.com/package/@jbrowse/cli) on NPM) that provides a way of downloading the latest JBrowse release and configuring it with assemblies, tracks, etc. You may be able to make use of that.

We're excited that you're interested in JBrowse 2! Please let us know if you have further questions.

**Reply by bbimber** [reply]
OK. I still have not 100% wrapped by head around reactJS or JB2. In rough terms, I need to download/build JB2 as part of our module's build process, and place the resulting javascript files into our webapp directory. this will package JB2 with our module. It seems like perhaps depending on the cli, and then executing the cli to download/build JB2 during our build might do this. I'll look more into it. 

At least with JBrowse 1, we packaged and included unconfigured JBrowse JS code. Our server would create the JSON config on-demand to initialize what we call "sessions", which is basically a configured JB instance. When I looked at the code in JB2, it seemed like I would be able to create new instances of the browser in a webpage, and configure that instance with a URL to the JSON file that specified the assembly, tracks, etc. Again, we need to look at this more.

**Reply by rbuels** [reply]
There is also a non-react embeddable linear genome view available as well, see the "JB2 Embedding" course at https://jbrowse.org/jb2/docs/bcc2020_embedding_jbrowse_01_getting_started
